bme680_bsec: Add information and example about multiple devices support

bisbastuner opened the pull request:
## Description:
PR 3550 (https://github.com/esphome/esphome/pull/3550) adds support for multiple devices, so explicitly state that, and add an example configuration that uses multiple devices

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#<esphome PR number goes here>

## Checklist:

  - [ ] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

bisbastuner committed:
Add information and example about multiple devices

PR 3550 (https://github.com/esphome/esphome/pull/3550) adds support for multiple devices, so explicitly state that, and add an example configuration that uses multiple devices

trvrnrth committed:
Standardise on I²C

trvrnrth committed:
Format possible addresses as code

trvrnrth commented:
I've done some very minor nit-picking in https://github.com/bisbastuner/esphome-docs/pull/1 but this looks good to me 👍

bisbastuner committed:
Update statements on number of supported devices

Latest code identifies devices by an index, and that allows to support a theoretical infinite number of devices.

ssieb commented:
This needs to target `next`, not `current`.
I don't see the point of including that huge config.  It's not any different than the other components that can have multiple devices.  What needs to be documented is the `bme680_bsec_id:` parameter.

bisbastuner commented:
> This needs to target `next`, not `current`. I don't see the point of including that huge config. It's not any different than the other components that can have multiple devices. What needs to be documented is the `bme680_bsec_id:` parameter.

Hmm.. I think this means that it would be better to document that by extending the example configuration above, what do you think?

And about the target branch, should I open a new PR and close this or should I rebase this PR?

ssieb commented:
> I think this means that it would be better to document that by extending the example configuration above, what do you think?

I don't think you need to change any examples, but you could if you want.  Just document the `bme680_bsec_id:` parameter in both sections.  See the Dallas component for example.

ssieb commented:
> And about the target branch, should I open a new PR and close this or should I rebase this PR?

Rebase it and then use the edit button at the top of this page to change the target branch.

bisbastuner committed:
Document id and bme680_bsec_id parameters

The bme680_bsec/id and sensor/bme680_bsec_id parameters are used when working with multiple devices: document them.

bisbastuner commented:
@ssieb I think I did it. Please tell me if I did something wrong, these are my first contributions!

trvrnrth commented on components/sensor/bme680_bsec.rst:
```diff
@@ -163,6 +169,13 @@ For each sensor all other options from :ref:`Sensor <config-sensor>` and :ref:`T
+        # id
+        # -----------
+        # Identifier for this component, useful when working with multiple devices.
+        # Muse be unique, and can be used in the sensor sections to refer to the correct device.
```
```suggestion
        # Must be unique, and can be used in the sensor sections to refer to the correct device.
```

bisbastuner committed:
Update components/sensor/bme680_bsec.rst

Fix typo

Co-authored-by: Trevor North <trevor@freedisc.co.uk>

trvrnrth commented:
This isn't stale. Just needs final maintainer review.

nagyrobi committed:
Merge branch 'next' into patch-1

nagyrobi committed:
Merge branch 'next' into patch-1
